lib/resources/shim-head.js

```diff
@@ -2,6 +2,25 @@
 ((win) => {
   win.ELECTRON = true;
 
+  if (process.env.ELECTRON_PROTOCOL_SERVE_INDEX && window.location.protocol !== 'file:') {
```

Comment:
I suppose it does include a check for `process.env.ELECTRON_PROTOCOL_SERVE_INDEX` so this should only apply when using the custom protocol(s). Would it make it to messy to check against a list of well-known protocols?
e.g. replace `window.location.protocol !== 'file:'` with `isCustomProtocol(window.location.protocol)`
```
const nonCustomProtocols = ['http:', 'https:', 'ftp:', 'file:']; 
function isCustomProtocol(protocol) {
    return nonCustomProtocols.indexOf(protocol) === -1;
}
```